sqldelight-gradle-plugin/src/test/integration-hsql/src/main/sqldelight/com/squareup/sqldelight/hsql/integration/HsqlTest.kt

```diff
@@ -0,0 +1,34 @@
+package com.squareup.sqldelight.hsql.integration
+
+import com.google.common.truth.Truth.assertThat
+import com.squareup.sqldelight.sqlite.driver.JdbcDriver
+import java.sql.DriverManager
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+
+class HsqlTest {
+  val conn = DriverManager.getConnection("jdbc:hsqldb:mem:mymemdb")
+  val driver = object : JdbcDriver() {
+    override fun getConnection() = conn
+  }
+  val database = MyDatabase(driver)
+
+  @Before fun before() {
+    MyDatabase.Schema.create(driver)
+  }
+
+  @After fun after() {
+    conn.close()
+  }
+
+  @Test fun simpleSelect() {
+    database.dogQueries.insertDog("Tilda", "Pomeranian", true)
+    assertThat(database.dogQueries.selectDogs().executeAsOne())
+        .isEqualTo(Dog.Impl(
+            name = "Tilda",
+            breed = "Pomeranian",
+            is_good = true
```

Comment:
Should this file have tests that cover the other data types? I modeled it after the tests I saw for postgres, but admit that I'm not sure about the desired coverage.

Reply:
i think this is probably good, we can add more in as we see fit but this will make sure theres a test thats actually hitting a hsqldb